snapshot/git/gitdb/db.go

```diff
@@ -108,6 +131,7 @@ type stringAndError struct {
 // Checkout puts the value identified by id in the local filesystem, returning
 // the path where it lives or an error.
 func (db *DB) Checkout(id snapshot.ID) (path string, err error) {
+	db.workTreeLock.Lock()
```

Comment:
Yup, eventually we want to do something smarter. We might also be able to use multiple work-trees if that feature shows up.